GDJS/Runtime/ResourceLoader.ts

```diff
@@ -0,0 +1,566 @@
+     * this scene will be the next to be loaded.
+     */
+    private _prioritizeScene(sceneName: string): SceneLoadingTask | null {
+      console.log('Prioritize scene: ' + sceneName);
+      const taskIndex = this._sceneToLoadQueue.findIndex(
+        (task) => task.sceneName === sceneName
+      );
+      if (taskIndex < 0) {
+        // The scene is already loaded.
+        return null;
+      }
+      const task = this._sceneToLoadQueue[taskIndex];
+      this._sceneToLoadQueue.splice(taskIndex, 1);
+      this._sceneToLoadQueue.push(task);
+      return task;
+    }
+
+    private async _processResource(resource: ResourceData): Promise<void> {
+      const resourceManager = this._resourceManagersMap.get(resource.kind);
+      if (!resourceManager) {
+        logger.warn(
+          'Unknown resource kind: "' +
+            resource.kind +
+            '" for: "' +
+            resource.name +
+            '".'
+        );
+        return;
+      }
+      await resourceManager.processResource(resource.name);
+    }
+
+    getSceneLoadingProgress(sceneName: string): float {
+      return sceneName === this.currentLoadingSceneName
+        ? this.currentSceneLoadingProgress
+        : this.areSceneAssetsLoaded(sceneName)
+        ? 1
+        : 0;
+    }
+
+    /**
+     * @returns true when all the resources of the given scene are loaded
+     * (but maybe not parsed).
+     */
+    areSceneAssetsLoaded(sceneName: string): boolean {
+      return !this._sceneNamesToLoad.has(sceneName);
+    }
+
+    /**
+     * @returns true when all the resources of the given scene are loaded and
+     * parsed.
+     */
+    areSceneAssetsReady(sceneName: string): boolean {
+      return !this._sceneNamesToMakeReady.has(sceneName);
+    }
+
+    private _setSceneAssetsLoaded(sceneName: string): void {
+      this._sceneNamesToLoad.delete(sceneName);
+    }
+
+    private _setSceneAssetsReady(sceneName: string): void {
+      this._sceneNamesToMakeReady.delete(sceneName);
+    }
+
+    getResource(resourceName: string): ResourceData | null {
+      return this._resources.get(resourceName) || null;
+    }
+
+    // Helper methods used when resources are loaded from an URL.
+
+    /**
+     * Complete the given URL with any specific parameter required to access
+     * the resource (this can be for example a token needed to access the resource).
+     */
+    getFullUrl(url: string) {
+      const { gdevelopResourceToken } = this._runtimeGame._options;
+      if (!gdevelopResourceToken) return url;
+
+      if (!checkIfIsGDevelopCloudBucketUrl(url)) return url;
+
+      return addSearchParameterToUrl(
+        url,
+        'gd_resource_token',
+        encodeURIComponent(gdevelopResourceToken)
+      );
+    }
+
+    /**
+     * Return true if the specified URL must be loaded with cookies ("credentials")
+     * sent to grant access to them.
+     */
+    checkIfCredentialsRequired(url: string) {
```

Comment:
private

Reply:
Actually it's not used by the loader. It was in RuntimeGame initially. It should probably be in a `ResourceLoaderHelper`, but I'm not sure it worth creating an instance just for this.

Reply:
Let's not create a new class/file just for this indeed.